File: src/main/java/seremis/geninfusion/block/BlockConnectedGlass.java
package seremis.geninfusion.block;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.block.material.Material;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.util.IIcon;
import net.minecraft.world.IBlockAccess;
import seremis.geninfusion.helper.ConnectedTextureHelper;
import seremis.geninfusion.lib.Blocks;

public class BlockConnectedGlass extends GIBlock {

    private ConnectedTextureHelper textureHelper = new ConnectedTextureHelper(this, Blocks.CONNECTED_GLASS_UNLOCALIZED_NAME);

    public BlockConnectedGlass(Material material) {
        super(material);
        setBlockName(Blocks.CONNECTED_GLASS_UNLOCALIZED_NAME);
        setHardness(2F);
        setResistance(3F);
    }

    @Override
    @SideOnly(Side.CLIENT)
    public void registerBlockIcons(IIconRegister iconregister) {
        super.registerBlockIcons(iconregister);
        textureHelper.registerBlockIcons(iconregister);
    }

    @Override
    @SideOnly(Side.CLIENT)
    public IIcon getIcon(IBlockAccess blockAccess, int x, int y, int z, int side) {
        return textureHelper.getIcon(blockAccess, x, y, z, side);
    }

    @Override
    public boolean isOpaqueCube() {
        return false;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public boolean shouldSideBeRendered(IBlockAccess blockaccess, int x, int y, int z, int side) {
        return blockaccess.getBlock(x, y, z) != this && super.shouldSideBeRendered(blockaccess, x, y, z, side);
    }

    @Override
    public boolean renderAsNormalBlock() {
        return false;
    }
}

File: src/main/java/seremis/geninfusion/block/BlockConnectedGlass.scala
package seremis.geninfusion.block

import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.block.material.Material
import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.util.IIcon
import net.minecraft.world.IBlockAccess
import seremis.geninfusion.helper.ConnectedTextureHelper
import seremis.geninfusion.lib.Blocks

class BlockConnectedGlass(material: Material) extends GIBlock(material) {

    setUnlocalizedName(Blocks.ConnectedGlassName)
    setHardness(2F)
    setResistance(3F)

    val textureHelper: ConnectedTextureHelper = new ConnectedTextureHelper(this, Blocks.ConnectedGlassName)

    @SideOnly(Side.CLIENT)
    override def registerIcons(iconRegister: IIconRegister) {
        super.registerIcons(iconRegister)
        textureHelper.registerIcons(iconRegister)
    }

    @SideOnly(Side.CLIENT)
    override def getIcon(blockAccess: IBlockAccess, x: Int, y: Int, z: Int, side: Int): IIcon = textureHelper.getIcon(blockAccess, x, y, z, side)

    override def isOpaqueCube: Boolean = false

    @SideOnly(Side.CLIENT)
    override def shouldSideBeRendered(blockAccess: IBlockAccess, x: Int, y: Int, z: Int, side: Int): Boolean = blockAccess.getBlock(x, y, z) != this && super.shouldSideBeRendered(blockAccess, x, y, z, side)

    override def renderAsNormalBlock(): Boolean = false
}